labconnect/main/auth_routes.py

```diff
@@ -70,36 +75,48 @@ def saml_callback():
     errors = auth.get_errors()
 
     if not errors:
+        registered = True
         user_info = auth.get_attributes()
         # user_id = auth.get_nameid()
 
         data = db.session.execute(db.select(User).where(User.email == "email")).scalar()
 
         # User doesn't exist, create a new user
         if data is None:
-
-            # TODO: add data
-            user = User(
-                # email=email,
-                # first_name=first_name,
-                # last_name=last_name,
-                # preferred_name=json_request_data.get("preferred_name", None),
-                # class_year=class_year,
-            )
-
-            db.session.add(user)
-            db.session.commit()
-
+            registered = False
         # Generate JWT
         # token = create_access_token(identity=[user_id, datetime.now()])
-        code = generate_temporary_code(user_info["email"][0])
+        code = generate_temporary_code(user_info["email"][0], registered)
 
         # Send the JWT to the frontend
         return redirect(f"{current_app.config['FRONTEND_URL']}/callback/?code={code}")
 
     return {"errors": errors}, 500
 
 
+@main_blueprint.post("/register")
+@jwt_required()
+def registerUser():
+
+    user_id = get_jwt_identity()
+
+    # Gather the new user's information
+    json_data = request.get_json()
+    user = User(
+        email=user_id,
+        first_name=json_data.get("first_name"),
+        last_name=json_data.get("last_name"),
+        preferred_name=json_data.get("preferred_name"),
+        class_year=json_data.get("class_year"),
+        profile_picture=json_data.get("profile_pictures"),
+        website=json_data.get("website"),
+        description=json_data.get("description"),
+    )
+    db.session.add(user)
+    db.session.commit()
```

Comment:
https://github.com/LabConnect-RCOS/LabConnect-Backend/blob/main/docs/database_docs/Labconnect_DB.png
There are some additional rows that should be created, for example management_permissions, and user_department, user_major